### ISSUE TITLE
Draggable Button does not move the absolute position of its pointer/gesture listeners

### ISSUE THREAD

**Issue by afandria** [domokit/sky_engine#558]
Possibly related? #56 

I was following the [Widgets](https://github.com/domokit/sky_engine/blob/master/sky/packages/sky/lib/widgets/README.md) tutorial and extended MyButton to have onGestureScrollStart and onGestureScrollUpdate handlers. The goal was to have the button be draggable.

I accumulated the `dx` and `dy` values to transform the button.

```
  Transform makeTransformedButton() {
    return new Transform(
      child: new MyButton(
        child: new Text('Engage'),
        onPressed: _handleOnPressedCallback,
        onScrollStart: _handleOnScrollStartCallback,
        onScrollUpdate: _handleOnScrollUpdateCallback
      ),
      transform: new vector_math.Matrix4.identity().translate(-dx, dy)
    );
  }
```

__Side Note__: I had to use -dx instead of dx to make it follow my finger properly.

__Observed Behavior__: I can only drag/tap the button if I do so from its original position. Drags/taps from the button's visual position are ignored (unless the position overlaps with the original one).
__Expected Behavior__: Drag/tap handlers move with the Widget.

**Comment by abarth** [comment]
You add post your whole `main.dart` file to this bug?  I'm trying to reproduce it.

**Comment by abarth** [comment]
s/You/Can you/  :)

**Comment by afandria** [comment]
Sure, I simplified it to this: (a gigantic magenta draggable button with click counter).
Let me know if you need anything else.

```
import 'package:sky/widgets/basic.dart';

import 'dart:sky' as sky;
import 'package:vector_math/vector_math.dart' as vector_math;

class HelloWorldApp extends App {
  int counter = 0;

  // Accumulators for the current scroll.
  double dx = 0.0;
  double dy = 0.0;

  Transform makeTransform() {
    return new Transform(
      child: new MyButton(
        child: new Text('Draggable Button: ${counter}'),
        onPressed: _handleOnPressedCallback,
        onScrollStart: _handleOnScrollStartCallback,
        onScrollUpdate: _handleOnScrollUpdateCallback
      ),
      transform: new vector_math.Matrix4.identity().translate(-dx, dy)
    );
  }

  void _handleOnPressedCallback(sky.Event e) {
    setState(() {
      counter++;
    });
  }

  void _handleOnScrollStartCallback(sky.Event e) {}

  void _handleOnScrollUpdateCallback(sky.Event e) {
    setState(() {
      sky.GestureEvent ge = e as sky.GestureEvent;
      dx += ge.dx;
      dy += ge.dy;
    });
  }

  Widget build() {
    return makeTransform();
  }
}

final BoxDecoration _decoration = new BoxDecoration(
  backgroundColor: const Color(0xFFFF00FF),
  borderRadius: 5.0
);

class MyButton extends Component {
  final Widget child;
  final Function onPressed;
  final Function onScrollStart;
  final Function onScrollUpdate;

  MyButton({this.child, this.onPressed, this.onScrollStart, this.onScrollUpdate});

  Container makeContainer() {
    return new Container(
      height: 36.0,
      padding: const EdgeDims.all(8.0),
      margin: const EdgeDims.symmetric(horizontal: 8.0),
      decoration: _decoration,
      child: new Center(
        child: this.child
      )
    );
  }

  Widget build() {
    return new Listener(
      onGestureTap: (e) {
        if (this.onPressed != null) {
          this.onPressed(e);
        }
      },
      onGestureScrollStart: (e) {
        if (this.onScrollStart != null) {
          this.onScrollStart(e);
        }
      },
      onGestureScrollUpdate: (e) {
        if (this.onScrollUpdate != null) {
          this.onScrollUpdate(e);
        }
      },
      child: makeContainer()
    );
  }
}

void main() {
  runApp(new HelloWorldApp());
}
```

**Comment by abarth** [comment]
thanks!

**Comment by abarth** [comment]
I can repro the issue.  Studying now.

**Comment by abarth** [comment]
The fix is in #583

Also, instead of using scrollstart / scrollupdate to move the box around, you probably want to use onPointerMove to move the box.

**Comment by afandria** [comment]
Thanks!
